yrs/src/block.rs

```diff
@@ -498,12 +498,15 @@ impl Block {
     ) {
         match self {
             Block::Item(item) => {
+                let mut info = item.info();
                 let origin = if offset > 0 {
                     Some(ID::new(item.id.client, item.id.clock + offset - 1))
                 } else {
                     item.origin
                 };
-                let info = item.info();
+                if origin.is_some() {
+                    info |= HAS_ORIGIN;
```

Comment:
This is the main issue - we were using native `Item::info` method. If that item didn't have left origin (eg. it was first block) `HAS_ORIGIN` flag was not set. However if we encode block with offset > 0, we were encoding that left origin, but flag was not set, causing failures during deserialization.